src/VirtoCommerce.CatalogModule.Data/ExportImport2/ProductExportPagedDataSource.cs

```diff
@@ -64,12 +72,18 @@ protected virtual IEnumerable<IExportable> ToExportable(IEnumerable<ICloneable>
         {
             var models = objects.Cast<CatalogProduct>();
 
-            var exportableProducts = models.Select(x =>
+            var exportableProductsTasks = models.Select(async x =>
             {
                 var exportableProduct = AbstractTypeFactory<ExportableProduct>.TryCreateInstance().FromModel(x);
+
+                var searchResult = await _configurationSearchService.SearchNoCloneAsync(new ProductConfigurationSearchCriteria { ProductId = exportableProduct.Id });

```

Comment:
Use AbstractTypeFactory when creating ProductConfigurationSearchCriteria.
Dont use SearchNoCloneAsync because this data source cannot guarantee that configurtion object will not be changed.
Try to reduce the number of calls to SearchAsync().